src/charm.py

```diff
@@ -211,6 +207,24 @@ def _hydra_service_is_running(self) -> bool:
             return False
         return service.is_running()
 
+    @property
+    def _public_url(self):
+        # TODO: The else part does not make much sense, remove it
```

Comment:
Any idea what could we do instead? We could defer rendering the conf file until the urls are ready, but I find it error prone.

Reply:
Not sure tbh. This falls under the whole discussion of status handling. I think it would make most sense to return `None` in the `else` part, which would cause the charm to block unless some `dev` flag is set which would enable us to test it without having to setup the whole bundle.

Reply:
@gruyaume what do you think?

Reply:
- BlockedStatus makes sense if we're waiting for someone to take action. In this case, we're waiting on the ingress charm (traefik) to provide the URL's right? So I think having the charm in WaitingStatus makes more sense.
- I don't know about deferring. Does the lib for traefik has a custom event for informing that the url is available? If yes no need to defer, let's only leverage this event.

Reply:
WaitingStatus makes sense when the relation is there, but the url is not ready. 
If there is no relation (ie the user has not connected the 2 units) I think that it makes to block then, because they need to perform an action.

Reply:
Agreed